pkg/queue/handler_test.go

```diff
@@ -50,104 +48,74 @@ func TestHandlerBreakerQueueFull(t *testing.T) {
 		QueueDepth: 1, MaxConcurrency: 1, InitialCapacity: 1,
 	})
 	stats := network.NewRequestStats(time.Now())
-
 	h := ProxyHandler(breaker, stats, false /*tracingEnabled*/, blockHandler)
 
-	ctx, cancel := context.WithTimeout(context.Background(), 1500*time.Millisecond)
-	t.Cleanup(cancel)
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:8081/time", nil)
-	if err != nil {
-		t.Fatal("NewRequestWithContext =", err)
+	req := httptest.NewRequest(http.MethodGet, "http://localhost:8081/time", nil)
+	resps := make(chan *httptest.ResponseRecorder)
+	for i := 0; i < 3; i++ {
+		go func() {
+			rec := httptest.NewRecorder()
+			h(rec, req)
+			resps <- rec
+		}()
 	}
-	var (
-		eg      errgroup.Group
-		barrier sync.WaitGroup
-	)
-	barrier.Add(2)
-	eg.Go(func() error {
-		barrier.Done()
-		h(httptest.NewRecorder(), req)
-		return nil
-	})
-	eg.Go(func() error {
-		barrier.Done()
-		h(httptest.NewRecorder(), req)
-		return nil
-	})
-	barrier.Wait()
-	// Now we know the queue is full next should exit immediately.
-
-	eg.Go(func() error {
-		defer close(resp) // Make the other requests terminate.
-		rec := httptest.NewRecorder()
-		h(rec, req)
-		if got, want := rec.Code, http.StatusServiceUnavailable; got != want {
-			return fmt.Errorf("Code = %d, want: %d", got, want)
-		}
-		const want = "pending request queue full"
-		if got := rec.Body.String(); !strings.Contains(rec.Body.String(), want) {
-			return fmt.Errorf("Body = %q wanted to contain %q", got, want)
+
+	// The first we see should've failed.
```

Comment:
what causes the first request to fail?

Reply:
queue overflow (1 gets into the handler and is then held, 1 gets into the queue and 1 fails due to overflow)

Reply:
nm - reading this comment I thought the literal _first_ request would fail. When I guess the first request gets through and is blocked and the second is queued by the breaker. So it's the third request that's failing

Reply:
```suggestion
	// A request should fail because the queue is full
```